src/nvim/strings.c

```diff
@@ -45,6 +45,41 @@
 #include "nvim/os/os.h"
 #include "nvim/os/shell.h"
 
+/// Compares two strings for equality.
+bool strequal(const char *a, const char *b)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strcmp(a, b) == 0;
+}
+
+/// Compares two sized strings for equality.
+bool strnequal(const char *a, const char *b, size_t size)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strncmp(a, b, size) == 0;
+}
+
+/// Compares two strings for equality ignoring differences in case.
+bool striequal(const char *a, const char *b)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strcasecmp(a, b) == 0;
```

Comment:
@aktau the extra underscore is common practice in MSVC they mark the non underscore version as deprecated and throw a warning for the the original ones (even for [close, write, read](https://github.com/equalsraf/neovim/blob/tb-win32-any/src/nvim/os/win_defs.h#L55)) - more [here](http://stackoverflow.com/questions/23477746/what-are-the-posix-like-functions-in-msvcs-c-runtime) and a [full list](https://msdn.microsoft.com/en-us/library/ms235384.aspx).

As you said _stricmp seems to be the only way for Windows, assuming the locale is set. There is also a _stricmp_l(str, str, locale_t) and _mbsicmp.

Encoding correctness is trickier - most of the cases above compare strings from eval with string literals, the former use &encoding the later are compiler specific. Assuming everyone does the "right thing TM", I didnt expect this PR to tackle those - but maybe its time to add unittests for that.